src/encoded/ingestion/table_utils.py

```diff
@@ -0,0 +1,787 @@
+                elif row_idx == 1:
+                    date = row[1].split('=')[1].strip()
+                elif row_idx == 2:
+                    fields = self.process_fields(row)
+                else:
+                    break # we are done with this step
+        logger.info('Mapping table Version: %s, Date: %s\n' % (version, date))
+        logger.info('Mapping table fields: %s\n' % (", ".join(fields)))
+        return version, date, fields
+
+    def process_annotation_field_inserts(self):
+        """ Processes the annotation fields in the mapping table to produce inserts
+            Note that project and institution are required fields on the annotation
+            field schema and are not set here
+
+        :returns: list of annotation field inserts
+        """
+        inserts = []
+        with open(self.mapping_table, 'r', encoding='utf-8-sig') as f:
+            reader = csv.reader(f)
+            for row_idx, row in enumerate(reader):
+                insert = {}
+                if row_idx <= MappingTableHeader.HEADER_ROW_INDEX:  # skip header rows
+                    continue
+                for field_name, entry in zip(self.fields, row):
+                    if field_name not in self.annotation_field_schema['properties'] or not entry:
+                        continue
+                    if field_name in MappingTableHeader.INTEGER_FIELDS:  # handle int fields
+                        if entry is not None:  # entry=0 is a normal value
+                            insert[field_name] = int(entry)
+                    elif field_name in MappingTableHeader.BOOLEAN_FIELDS:  # handle bool fields
+                        if entry is not None:
+                            if entry == 'Y':
+                                insert[field_name] = True
+                            else:  # assume False if anything other than 'Y' is present
+                                insert[field_name] = False
+                    elif field_name in ['enum_list']:  # handle enum fields
+                        if entry is not None:
+                            field_type = row[self.FIELD_TYPE_INDEX]
+                            val_list = []
+                            if field_type == 'string':
+                                val_list = [en.strip() for en in entry.split(',') if en.strip()]
+                            elif field_type == 'number':
+                                val_list = [float(en.strip()) for en in entry.split(',') if en.strip()]
+                            elif field_type == 'integer':
+                                val_list = [int(en.strip()) for en in entry.split(',') if en.strip()]
+                            insert[field_name] = val_list
+                    else:  # handle all other fields with direct copy if they exist
+                        if field_name == 'pattern':  # must decode escape characters
+                            insert[field_name] = entry.encode().decode('unicode-escape')
```

Comment:
What a weird thing. I'm still mulling how I feel about this emotionally, but I get what you're saying. Is this really the recommended way?
```suggestion
                            insert[field_name] = entry.encode('utf-8').decode('unicode-escape')
```
I would make this into a separate function for independent testing, also probably for `dcicutils.misc_utils`. Count me as surprised that this function is not separately available somewhere already as part of unicode support. There is surely a more optimal way to do it that just involves scanning for slashes and only processing those.